Cargo.toml

```diff
@@ -18,6 +18,7 @@ futures = "0.3"
 get_chunk = { version = "1.2", features = ["stream", "size_format"] }
 git2 = "0.19"
 lazy_static = "1"
+libc = "0.2.169"
```

Comment:
Honestly, if possible, I'd rather prefer just shelling out to ulimit or /sys/procs/something/fd 

Additionally! Should be feature and OS gated

Reply:
E.g. `/proc/self/` has a lot of useful info

Reply:
/proc/self/limits